### PR TITLE
Add python parameter hints

### PR DESCRIPTION
**Pull request by jeremyrsmith**
- Shows you parameter hints for the function you're calling
- Shows you the docstring for the function you're calling

Not sure if giant docstrings would be an issue here.

### CODE REVIEW

File: polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala [hunk 1]
```diff
@@ -456,40 +457,47 @@ class PythonInterpreter(val kernelContext: KernelContext, dependencyProvider: De
   // want to avoid re-populating globals at every completion request. We'll only do it when the cell we're completing changes.
   private var lastCompletionCell = -1
 
+  private def populateCodeAndGlobals(cellContext: CellContext, code: String, pos: Int): (Int, Int) = {
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: might be useful to indicate that this is specifically for completions?

File: polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala [hunk 2]
```diff
@@ -524,7 +532,34 @@ class PythonInterpreter(val kernelContext: KernelContext, dependencyProvider: De
     cellContext: CellContext,
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: Ugh, github won't let me comment on its line but looks like we can remove `// TODO: can parameter hints be implemented for python?` :)

File: polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala [hunk 3]
```diff
@@ -524,7 +532,34 @@ class PythonInterpreter(val kernelContext: KernelContext, dependencyProvider: De
...
+      val docString = Try(sig.getAttr("docstring", classOf[PyCallable]).call())
+        .map(_.asInstanceOf[String])
+        .toOption.filterNot(_.isEmpty).map(ShortString.truncate)
+
+      val name = s"${sig.getAttr("name", classOf[String])}(${params.mkString(", ")})"
+      val hints = ParameterHints(
+        name,
+        docString,
+        params.toList.map {
+          name => ParameterHint(name, "", None) // TODO: can individual params have docstrings?
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: Hm, can we at least get the type of the param to show?

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: What type? 😛 This is python.

I couldn't figure out how to get info from jedi about type hints if there are any, but I'll play around with it a bit.

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: Turns out that after upgrading jedi, you can get type names through the `infer` method (for some reason?). Added that in.

Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala]: Nice!

**Pull request merged**